src/main/kotlin/com/craftschemers/hub/Hub.kt

```diff
@@ -3,11 +3,28 @@ package com.craftschemers.hub
 import org.bukkit.plugin.java.JavaPlugin
 
 class Hub : JavaPlugin() {
+
+    /*
+    This is the main entry-point for our plugin. Let's try to abstract away most of the code into their
+    respective classes, which we will reference throughout the plugin.
+     */
+
+    companion object {
+        lateinit var plugin: Hub
+    }
+    val playerManger = HubPlayerManager()
+
     override fun onEnable() {
-        // onEnable
+        plugin = this
+        setupMinigames()
     }
 
     override fun onDisable() {
-        // onDisable
+        // save to db
     }
+
+    private fun setupMinigames() {
+        this.server.pluginManager.registerEvents(Events(), this)
```

Comment:
```suggestion
        server.pluginManager.registerEvents(Events(), this)
```
`this` is unnecessary here.